src/main/scala/org/apache/spark/sql/hive/llap/LlapExternalCatalog.scala

```diff
@@ -91,6 +92,32 @@ private[spark] class LlapExternalCatalog(
     }
   }
 
+  override def createTable(
+      tableDefinition: CatalogTable,
+      ignoreIfExists: Boolean): Unit = {
+    logInfo(tableDefinition.toString)
+
+    assert(tableDefinition.identifier.database.isDefined)
+    val db = tableDefinition.identifier.database.get
+    requireDbExists(db)
+
+    if (tableExists(db, tableDefinition.identifier.table)) {
+      if (ignoreIfExists) {
+        // No-op
+      } else {
+        throw new TableAlreadyExistsException(db = db, table = tableDefinition.identifier.table)
+      }
+    } else {
+      val sessionState =
+        SparkSession.getActiveSession.get.sessionState.asInstanceOf[LlapSessionState]
+      val stmt = sessionState.connection.createStatement()
+      // Check the privilege by creating a dummy table with the given name.
+      stmt.executeUpdate(s"CREATE TABLE ${tableDefinition.identifier.quotedString} (dummy INT)")
+      super.dropTable(db, tableDefinition.identifier.table, true, true)
```

Comment:
Yes. Initially, I tried to reconstruct `CREATE TABLE` statement including table name, schema, partitions, storage format. However, it turns out that it was a endless and error prone approach. Also, Spark transform SQL into `CatalogTable` by not one-to-one mapping. We cannot reconstruct the real complex table. In addition to that, we need to prepare Spark 2.2 (that is also different from 2.1, again). This implementation is transparent to the underlying Spark changes without any loss of information.